csharp/src/Ice/Communicator.cs

```diff
@@ -154,11 +155,13 @@ public partial interface Communicator : global::System.IDisposable
         /// </summary>
         ///  <param name="name">The object adapter name.
         ///  </param>
+        ///  <param name="authenticationOptions">The authentication options used by the ssl transport, pass null if
+        ///  the adapter doesn't have any secure endpoints.</param>
         /// <returns>The new object adapter.
         ///  </returns>
 
         [global::System.CodeDom.Compiler.GeneratedCodeAttribute("slice2cs", "3.7.10")]
-        ObjectAdapter createObjectAdapter(string name);
+        ObjectAdapter createObjectAdapter(string name, SslServerAuthenticationOptions authenticationOptions = null);
```

Comment:
Would be nice to cleanup this file in a separate clean-up PR.

We should also merge CommunicatorI into Communicator like we did in C++.

Reply:
The parameter should be called serverAuthenticationOptions.
When in doubt, use the IceRPC C# name:

https://github.com/icerpc/icerpc-csharp/blob/a1b10df451c97fd0ef21049da939dab5268818af/src/IceRpc/Server.cs#L155